pkgs/development/python-modules/unsloth/default.nix

```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  setuptools-scm,
+  packaging,
+  numpy,
+  torch,
+  bitsandbytes,
+}:
+
+buildPythonPackage rec {
+  pname = "unsloth";
+  version = "2025-02-v2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "unslothai";
+    repo = "unsloth";
+    tag = "${version}";
+    hash = "sha256-Z6plMoaay5SU2AlU68T7F7C9i8Zr12C+rNA0BHH1dQ0=";
+  };
+
+  build-system = [ setuptools ];
+  nativeBuildInputs = [ setuptools-scm ];
+
+  dependencies = [
+    packaging
+    numpy
+    torch
+    bitsandbytes
+  ];
+
+  # The source repository contains no test
+  doCheck = false;
```

Comment:
Can you also add `pythonImportsCheck`?

Reply:
Importing requires a GPU, else an error is raised. I added a comment.